### PR TITLE
Fsh/rewrite

### PR DESCRIPTION
**Pull request by fharding1**
Rewrote like a lot of stuff. Also I messed up and included a ton of old commits and idk how to fix that.

### CODE REVIEW

File: cmd/scouting-backend/main.go [hunk 1]
```diff
@@ -16,8 +16,8 @@ import (
 // PG_PORT: postgres port
```
Comment (BrendanBurkhart) [inline cmd/scouting-backend/main.go]: We're going to need to make sure this stays in sync with the readme

File: cmd/scouting-backend/main.go [hunk 2]
```diff
@@ -44,14 +44,13 @@ func main() {
 		os.Exit(1)
```
Comment (BrendanBurkhart) [inline cmd/scouting-backend/main.go]: Would this be better as `"unable to connect to postgres server: %v\n"`? I keep reading this as if the error was the options...

File: internal/server/server.go [hunk 3]
```diff
@@ -106,541 +120,60 @@ func (s *Server) Run(httpAddr, httpsAddr string) error {
...
 
-	respond.JSON(w, map[string]string{"jwt": ss})
+	return router
 }
 
-func (s *Server) getUser(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-
-	username := vars["username"]
+const tbaURL = "http://www.thebluealliance.com/api/v3"
```
Comment (BrendanBurkhart) [inline internal/server/server.go]: Would this be better as an argument to server? I guess we're not going to change versions of the api anytime soon, so it probably doesn't matter.

File: cmd/scouting-backend/main.go (hunk 1 above)
Comment (fharding1) [reply cmd/scouting-backend/main.go]: Yeah IK it's kind of annoying we might just remove it.

File: cmd/scouting-backend/main.go (hunk 2 above)
Comment (fharding1) [reply cmd/scouting-backend/main.go]: Yeah. That's a bit confusing. Will fix 👍

File: internal/server/server.go (hunk 3 above)
Comment (fharding1) [reply internal/server/server.go]: If we changed API versions it would (probably) break our code, so imo it should be a constant.

Comment (BrendanBurkhart) [reply internal/server/server.go]: sg, that makes sense

**Comment by fharding1** [comment]
I forgot to update the API for the schema route will do that when I get home.

**Comment by fharding1** [comment]
*API docs

**Comment by calebeby** [comment]
Can we put titles for the fields in the schema file as well? That way the frontend knows what to display

**Comment by fharding1** [comment]
What do you mean by titles?

**Comment by calebeby** [comment]
Human readable titles, like Buckets Knocked Over or Crossed Midline in Auto

**Comment by fharding1** [comment]
Hmm. Could you just transform the "bucketsMoved" or whatever to "Buckets
Moved"? If not, I can make something for human readable titles but it's
gonna be weird.

**Comment by calebeby** [comment]
Yeah I can transform them, that's fine

**Comment by fharding1** [comment]
🙏

**Pull request merged**